src/graph/entity_extractor.py

```
@@ -0,0 +1,698 @@
+            chunks[i : i + self.config.batch_size]
+            for i in range(0, len(chunks), self.config.batch_size)
+        ]
+
+        with ThreadPoolExecutor(max_workers=self.config.max_workers) as executor:
+            futures = {executor.submit(self._extract_from_batch, batch): batch for batch in batches}
+
+            for future in as_completed(futures):
+                try:
+                    batch_entities = future.result()
+                    all_entities.extend(batch_entities)
+                except Exception as e:
```

Ruff (0.14.6)

Do not catch blind exception: `Exception`

(BLE001)